### PR TITLE
FontManager implementation

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?
This introduces a font manager that is responsible for managing installed fonts and the default font. This also introduces the new GlyphTypeface that is needed for GlyphRun etc.

## What is the current behavior?
Currently, there is no way to know which font is loaded as the default because we use an empty string.

## What is the updated/expected behavior with this PR?
It will be possible to know the default font at runtime. The font manager also introduces font fallback that will be used by the new text layout.

## How was the solution implemented (if it's not obvious)?
There is a new interface `IFontManagerImpl` that should be implemented per platform. Currently, there is one implementation for Direct2D1 and one for Skia.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes

`FontSize` is no longer part of the `Typeface`

### CODE REVIEW

File: src/Avalonia.Visuals/Media/GlyphTypeface.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+
+using Avalonia.Platform;
+
+namespace Avalonia.Media
+{
+    public class GlyphTypeface : IDisposable
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/GlyphTypeface.cs]: This just wrapps `IGlyphTypefaceImpl` so it should be safe to use the interface directly.

File: src/Avalonia.Visuals/Media/Typeface.cs [hunk 2]
```diff
@@ -1,39 +1,38 @@
...
+using System.Diagnostics;
+using JetBrains.Annotations;
 
 namespace Avalonia.Media
 {
     /// <summary>
     /// Represents a typeface.
     /// </summary>
-    public class Typeface
+    [DebuggerDisplay("Name = {FontFamily.Name}, Style = {Style}, Weight = {Weight}")]
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/Typeface.cs]: Switch style and weight

File: src/Avalonia.Visuals/Media/FontManager.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        private static IFontManagerImpl GetPlatformImpl()
+        {
+            var platformImpl = AvaloniaLocator.Current.GetService<IFontManagerImpl>();
+
+            return platformImpl ?? new EmptyFontManagerImpl();
+        }
+
+        private class EmptyFontManagerImpl : IFontManagerImpl
+        {
+            public string DefaultFontFamilyName => "Arial";
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/FontManager.cs]: Call the default "Empty". All platforms should have an implementation of the manager. This is just to make testing easier.

File: src/Avalonia.Visuals/Media/FontFamily.cs [hunk 4]
```diff
@@ -5,12 +5,16 @@
 using System.Collections.Generic;
 using System.Linq;
 using Avalonia.Media.Fonts;
-using Avalonia.Platform;
 
 namespace Avalonia.Media
 {
     public class FontFamily
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/FontFamily.cs]: Seal this

File: src/Avalonia.Visuals/Media/Typeface.cs [hunk 5]
```diff
@@ -1,39 +1,38 @@
...
+    public class Typeface : IEquatable<Typeface>
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/Typeface.cs]: Seal this

File: src/Avalonia.Visuals/Media/Fonts/FamilyNameCollection.cs [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
 namespace Avalonia.Media.Fonts
 {
-    public class FamilyNameCollection : IEnumerable<string>
+    public class FamilyNameCollection : IReadOnlyList<string>
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/Fonts/FamilyNameCollection.cs]: Seal this

File: src/Skia/Avalonia.Skia/GlyphTypefaceImpl.cs [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+using System;
+using System.Runtime.InteropServices;
+using Avalonia.Media;
+using Avalonia.Platform;
+using HarfBuzzSharp;
+using SkiaSharp;
+
+namespace Avalonia.Skia
+{
+    internal class GlyphTypefaceImpl : IGlyphTypefaceImpl
```
Comment (Gillibald) [inline src/Skia/Avalonia.Skia/GlyphTypefaceImpl.cs]: Make this public

File: src/Windows/Avalonia.Direct2D1/Media/GlyphTypefaceImpl.cs [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+using System;
+using Avalonia.Media;
+using Avalonia.Platform;
+using HarfBuzzSharp;
+using SharpDX.DirectWrite;
+
+namespace Avalonia.Direct2D1.Media
+{
+    internal class GlyphTypefaceImpl : IGlyphTypefaceImpl
```
Comment (Gillibald) [inline src/Windows/Avalonia.Direct2D1/Media/GlyphTypefaceImpl.cs]: Make this public

File: build/SkiaSharp.props [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup>
-    <PackageReference Include="SkiaSharp" Version="1.68.0" />
-    <PackageReference Condition="'$(IncludeLinuxSkia)' == 'true'" Include="Avalonia.Skia.Linux.Natives" Version="1.68.0.2" />
+    <PackageReference Include="SkiaSharp" Version="1.68.1-rc.153" />
```
Comment (jmacato) [inline build/SkiaSharp.props]: @Gillibald  Just for reference, i wonder when can we expect a full release version of both HarfBuzzSharp and SkiaSharp?

**Review by jmacato — APPROVED** [review]
This looks great! No bugs/regressions from ControlCatalog.

File: build/SkiaSharp.props (hunk 9 above)
Comment (Gillibald) [reply build/SkiaSharp.props]: The plan is to have a release a month after the first preview. So if nothing breaks we get it until end of november.

**Pull request merged**